### ISSUE TITLE
Fatal signal 6 (SIGABRT)

### ISSUE THREAD

**Issue by MaddinMade** [isar/isar#411]
Unfortunately, I can't reproduce the bug nor do I know when it occurs, but for some users, the app has been crashing with this bug lately:

```
Scudo ERROR: invalid chunk state when deallocating address 0x0079061d9570
2022-06-16 19:49:33.129 31615-16695/com.example.app A/libc: Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 16695 (1.ui), pid 31615 (com.example.app)
2022-06-16 19:49:33.425 16808-16808/? A/DEBUG: pid: 31615, tid: 16695, name: 1.ui  >>> com.example.app <<<
2022-06-16 19:49:33.516 16808-16808/? A/DEBUG:       #05 pc 0000000000025154  /data/app/~~GLeuy0PEBcEePM3Q1j33DA==/com.example.app-TKpx-w1jYZa-9v4V0RgKDA==/lib/arm64/libisar.so (isar_txn_finish+372)
```

I will add details as soon as I know when exactly it occurs. Once it occurs, you have to reinstall the app and then it works again for a while.

**Version**
 - Platform: Android
 - Flutter version: 3.0.2
 - Isar version: 2.5.0
 - Isar Generator version: 2.5.0

**Comment by MaddinMade** [comment]
The error occurs in two places. Both use a composite index and filter for a string with equalTo and another value (int and DateTime) with lessThan. As soon as I don't use an index at these places any more, no error is thrown. However, since I use distinct in the one place, the transitional solution is relatively inefficient since I have to do it manually.

**Comment by leisim** [comment]
It would be incredibly helpful if you found a code sample that reproduces the issue somewhat reliably

**Comment by MaddinMade** [comment]
@leisim Yes, I try, but the app is big and the error always ist thrown after a while. So it is difficult to find the trigger.

**Comment by MaddinMade** [comment]
@leisim I will try again tomorrow. I tried to find the cause, but I only found the lines where the error occurs.

**Comment by MaddinMade** [comment]
@leisim I don't know if this has a similar cause, but I found the following comment: https://github.com/isar/isar/issues/296#issuecomment-1077489889
Was this problem also fixed or only the NullPointerException?

**Comment by leisim** [comment]
@MaddinMade the issue in the comment was caused by links. I already fixed the error causing it. A update will go out today or tomorrow 🙂

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.8`

**Comment by MaddinMade** [comment]
@leisim Sorry I couldn't provide any info. In which situations did the error occur? I just threw out all the indexes at some point because there was always a crash when using them after a while.
